app/models/post.rb

```diff
@@ -0,0 +1,21 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :likes, foreign_key: :post_id, dependent: :delete_all
+  has_many :comments, foreign_key: :post_id, dependent: :delete_all
+
+  after_save :update_post_counter
+
+  validates :title, presence: true
+  validates :comments_counter, numericality: { only_integer: true }, comparison: { greater_than_or_equal_to: 0 }
```

Comment:
- [ ] You forgot to add validation for the title that checks it must not exceed 250 characters. please make sure to add it  to complete the requirements